common/http/src/main/java/io/helidon/common/http/ByteBufferDataChunk.java

```diff
@@ -29,7 +30,7 @@ final class ByteBufferDataChunk implements DataChunk {
     private final boolean flush;
     private final boolean readOnly;
     private final Runnable releaseCallback;
-    private boolean isReleased = false;
+    private final AtomicBoolean isReleased = new AtomicBoolean();
```

Comment:
You might consider using `VarHandle` instead, so you aren't lugging around a full `AtomicBoolean` inside every single `DataChunk`, just one static field and a `volatile boolean` instead.